staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -346,6 +346,11 @@ type CustomResourceDefinitionCondition struct {
 	// Human-readable message indicating details about last transition.
 	// +optional
 	Message string
+	// observedGeneration represents the .metadata.generation that the condition was set based upon.
+	// For instance, if .metadata.generation is currently 12, but the .status.conditions[x].observedGeneration is 9, the condition is out of date
+	// with respect to the current state of the instance.
+	// +optional
+	ObservedGeneration int64
```

Comment:
add a gate to staging/src/k8s.io/apiextensions-apiserver/pkg/features/kube_features.go to help us roll this new field out, and document this is alpha gated on CRDObservedGenerationTracking

```
	CRDObservedGenerationTracking: {
		{Version: version.MustParse("1.35"), Default: false, PreRelease: featuregate.Alpha},
	},
```

in staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/strategy.go#dropDisabledFields, zero this out if the gate is off and the field is not already in use (see dropDisabledPodStatusFields for an example)